### PR TITLE
Initial work on filling gaps in versioning docs

### PR DESCRIPTION
**Pull request by monikasulik**

### CODE REVIEW

File: djangocms_versioning/admin.py [hunk 1]
```diff
@@ -31,17 +31,20 @@
...
 
     def get_queryset(self, request):
+        """Limit the content model queryset to latest versions only."""
         queryset = super().get_queryset(request)
         versionable = versionables.for_content(queryset.model)
         return queryset.filter(pk__in=versionable.distinct_groupers())
 
 
-def versioning_change_list_factory(ChangeList):
+def versioning_change_list_factory(BaseChangeListClass):
```
Comment (chaosk) [inline djangocms_versioning/admin.py]: Think if you add `class` to the name, then convert it to snake case (`base_change_list_cls`?)

File: docs/admin_architecture.rst [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+
+The Version model admin
+------------------------
+
+Proxy models
++++++++++++++
+Versioning generates a `proxy model
+<https://docs.djangoproject.com/en/dev/topics/db/models/#proxy-models>`_
+for each
```
Comment (chaosk) [inline docs/admin_architecture.rst]: a proxy model of Version for each content model?

File: docs/index.rst [hunk 3]
```diff
@@ -3,8 +3,52 @@ Welcome to "djangocms-versioning"'s documentation!
 
 .. toctree::
    :maxdepth: 3
-   :caption: Contents:
+   :caption: Introduction:
 
    versioning_integration
+
```
Comment (geoidesic) [inline docs/index.rst]: Doesn't look like you edited `versioning_integration.rst` however I'm getting a build warning when running `make html`:

```
/Users/noeldacosta/Documents/projects/fidelity/djangocms-versioning/docs/versioning_integration.rst:32: WARNING: Error in "code-block" directive:
maximum 1 argument(s) allowed, 7 supplied.

.. code-block:: python
    # blog/models.py
    from django.db import models


    class Post(models.Model):
        pass


    class PostContent(models.Model):
        post = models.ForeignKey(Post, on_delete=models.CASCADE)
        title = models.CharField(max_length=100)
        text = models.TextField()
```
Can this be fixed?

File: djangocms_versioning/admin.py [hunk 4]
```diff
@@ -104,6 +118,7 @@ def get_readonly_fields(self, request, obj=None):
         return super().get_readonly_fields(request, obj)
 
     def has_change_permission(self, request, obj=None):
+        # TODO: This is possibly a bug
```
Comment (geoidesic) [inline djangocms_versioning/admin.py]: TODO – if it is a bug please raise a ticket for it. Probably shouldn't include this comment in the release.

File: djangocms_versioning/admin.py [hunk 5]
```diff
@@ -192,7 +217,11 @@ def get_changelist(self, request, **kwargs):
         return VersionChangeList
 
     def get_list_filter(self, request):
+        """Adds the filters for the extra grouping fields to the UI."""
         versionable = versionables.for_content(self.model._source_model)
+        # TODO: This possibly should still allow adding more filters via
```
Comment (geoidesic) [inline djangocms_versioning/admin.py]: same here - add ticket and remove todo

File: djangocms_versioning/admin.py [hunk 6]
```diff
@@ -223,6 +252,8 @@ def get_list_display(self, request):
         )
 
     def get_actions(self, request):
+        """Removes the standard django admin delete action."""
+        # TODO: There appears to be no test for this.
```
Comment (geoidesic) [inline djangocms_versioning/admin.py]: same here - add ticket and remove todo

File: djangocms_versioning/admin.py [hunk 7]
```diff
@@ -667,6 +699,8 @@ def revert_view(self, request, object_id):
                 request, "djangocms_versioning/admin/revert_confirmation.html", context
             )
         else:
+            # TODO: What's going on here? Why are we allowing both the
+            # archiving of a draft and the deletion of it straight after?
```
Comment (geoidesic) [inline djangocms_versioning/admin.py]: I don't see that it does that. The logic forks here depending on GET parameters. It does either an 'archive' or 'discard'.

File: docs/index.rst [hunk 8]
```diff
@@ -2,10 +2,61 @@ Welcome to "djangocms-versioning"'s documentation!
...
+--------
+
+.. glossary::
+
+    version model
+        A model that stores information such as state (draft, published etc.),
+        author, created and modified dates etc. about each version.
+
+    content model
+        A model with a one2one relationship with the
```
Comment (geoidesic) [inline docs/index.rst]: I suggest add in the image that I put int the `docs/_static` folder which illustrates this.

Comment (geoidesic) [reply docs/index.rst]: See here: https://divio-ch.atlassian.net/wiki/spaces/FIL/pages/668041217/Gaps+Documentation#GapsDocumentation-djangocms-versioning(Divio)

Kunjal found this very helpful. Not sure if Glossary is the right place for it.

File: djangocms_versioning/datastructures.py [hunk 9]
```diff
@@ -151,6 +164,12 @@ def _get_content_types(self):
 
     @cached_property
     def content_types(self):
+        """Get the primary key of the content type of the registered content model.
+
+        :return:  A list with one element - the pk of the content type of the content model
+        """
+        # TODO: Why on earth would this return a list which will always
+        # have just one element?
```
Comment (chaosk) [inline djangocms_versioning/datastructures.py]: This won't always have just one element - see PolymorphicVersionableItem._get_content_types.

File: docs/admin_architecture.rst [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+include only the records for the latest versions.
+
+
+The Version model admin
+------------------------
+
+Proxy models
++++++++++++++
+Versioning generates a `proxy model
+<https://docs.djangoproject.com/en/dev/topics/db/models/#proxy-models>`_ of Version
```
Comment (chaosk) [inline docs/admin_architecture.rst]: Can you refer to `:class:djangocms_versioning.models.Version` here? Think it makes sense in versioning, as this documentation gets quite complex

File: docs/index.rst (hunk 8 above)
Comment (chaosk) [reply docs/index.rst]: Can use https://www.sphinx-doc.org/en/master/usage/extensions/graphviz.html for that

**Comment by chaosk** [comment]
Overall really well done, there are some minor comments left

File: djangocms_versioning/admin.py (hunk 7 above)
Comment (monikasulik) [reply djangocms_versioning/admin.py]:
```
            if draft_version and request.POST.get("archive"):
                draft_version.archive(request.user)
            if draft_version and request.POST.get("discard"):
                draft_version.delete()
```
Based on this, if a url had both archive and discard in the url, both of these would happen.

File: djangocms_versioning/datastructures.py (hunk 9 above)
Comment (monikasulik) [reply djangocms_versioning/datastructures.py]: aaah, thanks! Will correct the docstring/comment.

File: djangocms_versioning/admin.py (hunk 7 above)
Comment (geoidesic) [reply djangocms_versioning/admin.py]: Yes that's true, although I don't imagine that the UI would create such a scenario. What do you suggest?

Comment (monikasulik) [reply djangocms_versioning/admin.py]: Writing a test and changing the second if to an elif ;)

Comment (monikasulik) [reply djangocms_versioning/admin.py]: https://divio-ch.atlassian.net/browse/FIL-1003

File: docs/index.rst (hunk 8 above)
Comment (monikasulik) [reply docs/index.rst]: I'll have a go at adding a graphviz version of that diagram to the versioning integration page as that's really where I try to explain these concepts in more detail.

**Pull request merged**